code/datums/wires/taperecorder.dm

```diff
@@ -0,0 +1,28 @@
+/datum/wires/taperecorder
+	holder_type = /obj/item/taperecorder
+	proper_name = "Universal Recorder"
+	req_knowledge = JOB_SKILL_TRAINED
+	req_skill = JOB_SKILL_UNTRAINED
+
+/datum/wires/taperecorder/New(atom/holder)
+	wires = list(
+		WIRE_RECORD,
+		WIRE_STOP, WIRE_PLAY
+	)
+	..()
+
+/datum/wires/taperecorder/interactable(mob/user)
+	. = FALSE
+	var/obj/item/taperecorder/T = holder
+	if(T.panel_open)
+		. = TRUE
+
+/datum/wires/taperecorder/on_pulse(wire)
+	var/obj/item/taperecorder/T = holder
+	switch(wire)
+		if(WIRE_RECORD)
+			T.record()
+		if(WIRE_STOP)
+			T.stop()
+		if(WIRE_PLAY)
+			T.play()
```

Comment:
```suggestion
/datum/wires/taperecorder/on_pulse(wire)
	var/obj/item/taperecorder/recorder = holder
	switch(wire)
		if(WIRE_RECORD)
			recorder.record()
		if(WIRE_STOP)
			recorder.stop()
		if(WIRE_PLAY)
			recorder.play()

```
* Use informative variable names.
* Leave a trailing newline at the end of the file. The compiler will screech if you do not. This is about preserving git blame, and dealing with some parsers appending multiple DM files wrongly else.